specification/compute/resource-manager/Microsoft.Compute/stable/2022-01-03/GalleryRP/galleryRPCommon.json

```diff
@@ -124,6 +124,89 @@
         }
       },
       "description": "Describes the gallery image definition purchase plan. This is used by marketplace images."
+    },
+    "SharedGalleryImageVersionStorageProfile": {
+      "type": "object",
+      "properties": {
+        "osDiskImage": {
+          "$ref": "#/definitions/SharedGalleryOSDiskImage"
+        },
+        "dataDiskImages": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedGalleryDataDiskImage"
+          },
+          "x-ms-identifiers": [
+            "lun"
+          ],
+          "description": "A list of data disk images."
+        }
+      },
+      "description": "This is the storage profile of a Gallery Image Version."
+    },
+    "SharedGalleryOSDiskImage": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/SharedGalleryDiskImage"
+        }
+      ],
+      "description": "This is the OS disk image."
+    },
+    "SharedGalleryDataDiskImage": {
+      "type": "object",
+      "properties": {
+        "lun": {
+          "type": "integer",
+          "format": "int32",
+          "description": "This property specifies the logical unit number of the data disk. This value is used to identify data disks within the Virtual Machine and therefore must be unique for each data disk attached to the Virtual Machine."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/SharedGalleryDiskImage"
+        }
+      ],
+      "required": [
+        "lun"
+      ],
+      "description": "This is the data disk image."
+    },
+    "SharedGalleryDiskImage": {
+      "type": "object",
+      "properties": {
+        "diskSizeGB": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "description": "This property indicates the size of the VHD to be created."
+        },
+        "hostCaching": {
+          "type": "string",
+          "description": "The host caching of the disk. Valid values are 'None', 'ReadOnly', and 'ReadWrite'",
+          "enum": [
+            "None",
+            "ReadOnly",
+            "ReadWrite"
+          ],
+          "x-ms-enum": {
+            "name": "HostCaching",
+            "modelAsString": false
```

Comment:
>modelAsString [](http://example.com/codeflow?start=13&length=13)

Recommend modelAsString: true, since it supports adding new enum values without needing a new API version.

Reply:
addressed